test/test_utils.py

```diff
@@ -224,3 +224,18 @@ def test_invalid_png_data(self):
                           imageinfo.identify, data)
         self.assertRaises(imageinfo.UnrecognizedFormat,
                           imageinfo.identify, data)
+
+class CompareBarcodesTest(unittest.TestCase):
+
+    def test_same(self):
+        self.assertTrue(util.compare_barcodes('0727361379704', '0727361379704'))
+        self.assertTrue(util.compare_barcodes('727361379704', '727361379704'))
+        self.assertTrue(util.compare_barcodes('727361379704', '0727361379704'))
+        self.assertTrue(util.compare_barcodes('0727361379704', '727361379704'))
+        self.assertTrue(util.compare_barcodes(None, None))
+
+    def test_not_same(self):
+        self.assertFalse(util.compare_barcodes('0727361379704', '0727361379705'))
+        self.assertFalse(util.compare_barcodes('727361379704', '1727361379704'))
+        self.assertFalse(util.compare_barcodes('0727361379704', None))
+        self.assertFalse(util.compare_barcodes(None, '0727361379704'))
```

Comment:
I'd add tests for empty strings too:

```python
self.assertFalse(util.compare_barcodes(None, ''))
```

Reply:
Is this actually what we would expect, or shouldn't `None` and `''` be considered equal?

Reply:
I guess it'd make sense.

Reply:
Yes, I think it makes most sense. I implemented it this way, which needed some update to the compare_barcodes function itself.